eth/interop.go

```diff
@@ -17,6 +18,21 @@ func (s *Ethereum) CheckAccessList(ctx context.Context, inboxEntries []common.Ha
 	return s.interopRPC.CheckAccessList(ctx, inboxEntries, minSafety, execDesc)
 }
 
+func (s *Ethereum) inferBlockTime(current *types.Header) (uint64, error) {
+	penultimate := s.BlockChain().GetHeaderByHash(current.ParentHash)
+	if penultimate == nil {
+		// We could use a for loop and retry, but this function is used
+		// in the ingress filters, which should fail fast to maintain uptime.
+		if current.Number.Uint64() == 0 {
+			return 0, errors.New("current head is at genesis: penultimate header is nil")
+		} else {
+			return 0, errors.New("a reorg probably occurred: penultimate header is nil")
+		}
+	}
+	blockTime := current.Time - penultimate.Time
+	return current.Time + blockTime, nil
```

Comment:
I would have expected this function to return just the `blockTime` here, and let the caller add it to their `current` as needed, since it specifies it returns `BlockTime` (which to me is a duration). OR you can rename this function `inferNextBlockTime` to specify that it is adjusting *from* the given head *to* the next time.

Reply:
copy-pasta mistake, thanks.